### PR TITLE
Improve and stabilize system tests

### PR DESCRIPTION
**Pull request by krucod3**
Issues: #92

# Definition of Done

The PR shall be merged only if all items mentioned in 
[CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute)
have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: tests/resources/ankaios.resource [hunk 1]
```diff
@@ -87,125 +85,115 @@ Podman has deleted all existing volumes
...
+    &{dict_result}=    table to dict    ${list_result}    WORKLOAD NAME
+    &{item}=    Get From Dictionary    ${dict_result}    ${workload_name}
+    ${actual_execution_state}=    Get From Dictionary    ${item}    EXECUTION STATE
+    ${actual_agent_name}=    Get From Dictionary    ${item}    AGENT
+    Should be equal    ${actual_execution_state}    ${expected_execution_state}
+    Should be equal    ${actual_agent_name}    ${expected_agent_name}
 
-the workload "${workload_name}" shall not exist
-    Log    Assert: ${workload_name} shall not exist
+the workload "${workload_name}" shall not exist within "${timeout}" seconds
```
Comment (krucod3) [inline tests/resources/ankaios.resource:134]: It seems like we don't use the timeout here at all. Maybe we can remove it?

File: tests/resources/ankaios.resource [hunk 2]
```diff
@@ -87,125 +85,115 @@ Podman has deleted all existing volumes
...
-the JSON array "${dict_array}" shall contain key "${main_key}" with subkey "${sub_key}" with the subvalue "${sub_value}"
-    Log    ${dict_array}
-    FOR    ${dict_array_item}    IN    @{dict_array}
-        Log    ${dict_array_item}
-        Dictionary Should Contain Key    dictionary=${dict_array_item}    key=${main_key}
-        Dictionary Should Contain Item
-        ...    dictionary=${dict_array_item["${main_key}"]}
-        ...    key=${sub_key}
-        ...    value=${sub_value}
+volumes for "${workload_name}" shall exist on "${agent_name}" within "${timeout}" seconds
```
Comment (krucod3) [inline tests/resources/ankaios.resource:145]: The timeout is not used here.

File: tests/resources/ankaios.resource (hunk 1 above)
Comment (lingnoi) [reply tests/resources/ankaios.resource:129]: timeout is handled now

File: tests/resources/ankaios.resource (hunk 2 above)
Comment (lingnoi) [reply tests/resources/ankaios.resource:140]: timeout is handled now

**Comment by krucod3** [comment]
LGTM 👍 
Once the cache is fixed in #102 and get a green build here, we can merge this one.

**Review by lingnoi — APPROVED** [review]
LGTM 👍 :)

**Pull request merged**